src/cars/cars.controller.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  Controller,
+  Post,
+  Get,
+  Param,
+  Delete,
+  Body,
+  Put,
+  Query,
+} from '@nestjs/common';
+import { ApiBearerAuth, ApiQuery, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { CarsService } from './cars.service';
+import { CreateCarDto } from './dto/create-car.dto';
+import { Auth } from 'src/auth/decorators/auth.decorator';
+import { AuthUser } from 'src/auth/decorators/auth-user.decorator';
+import { CreateCarResponseDto } from './dto/create-car-response.dto';
+import { UpdateCarDto } from './dto/update-car.dto';
+import { AuthRoles } from 'src/common/enums/roles.enum';
+import { PageOptionsDto } from 'src/common/helpers/pagination/pagination.dtos';
+import { PageDto } from 'src/common/helpers/pagination/page.dto';
+import { Car } from './entities/car.entity';
+
+@ApiTags('cars')
+@Controller('cars')
+export class CarsController {
+  constructor(private readonly carsService: CarsService) {}
+  @Post()
+  @Auth()
+  @ApiBearerAuth()
+  @ApiResponse({
+    status: 201,
+    description: 'Car created successfully',
+    type: CreateCarResponseDto,
+  })
+  @ApiResponse({ status: 400, description: 'Bad Request' })
+  @ApiResponse({ status: 500, description: 'Internal Server Error' })
+  async createCar(
+    @Body() createCarDto: CreateCarDto,
+    @AuthUser() user: AuthUser,
```

Comment:
You can get the user directly from @Req if you pass it after validating the token.